### ISSUE TITLE
Fix request: false positive NonComparableMapKeys for Hazelcast maps

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#565]
Does this rule also make sense for hazelcast maps? 

```java
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.core.HazelcastInstance;
import com.hazelcast.map.IMap;
import com.hazelcast.map.LocalMapStats;

class Foo {
    public LocalMapStats getMapStats(final String cacheName) {
        final HazelcastInstance client = HazelcastClient.getHazelcastClientByName("my-cache");
        final IMap<Object, Object> map = client.getMap("my-map"); // <-- false positive
        return map.getLocalMapStats();
    }
}
```

Note there is a similar Sonar rule: [S6411](https://community.sonarsource.com/t/suggestion-to-add-a-sonarlint-rule-for-detecting-incorrect-map-keys/108466) that also creates an issue for this Hazelcast code.

**Comment by stokpop** [comment]
Let's close this issue until there is proof this is a false positive.
